src/Controller/ProgrammeController.php

```diff
@@ -24,38 +25,49 @@ class ProgrammeController
 
     private ProgrammeRepository $programmeRepository;
 
-    private int $maxProgrammesPerPage;
+    private ProgrammeRequestContentType $programmeRequestContentType;
 
     public function __construct(
         EntityManagerInterface $entityManager,
         SerializerInterface $serializer,
         ProgrammeRepository $programmeRepository,
-        string $maxProgrammesPerPage
+        ProgrammeRequestContentType $programmeRequestContentType
     ) {
         $this->entityManager = $entityManager;
         $this->serializer = $serializer;
         $this->programmeRepository = $programmeRepository;
-        $this->maxProgrammesPerPage = (int)$maxProgrammesPerPage;
+        $this->programmeRequestContentType = $programmeRequestContentType;
     }
 
     /**
      * @Route (methods={"GET"})
      */
     public function getAllProgrammes(Request $request): Response
     {
+        $contentSubtype = $this->programmeRequestContentType->getRequestType($request);
+        if ($contentSubtype !== 'json' && $contentSubtype !== 'xml') {
```

Comment:
think of a way to better separate the accepted types here. if I add 3 more accepted types, I'll get 3 more conditions. how can  you make this to be more dynamic?